### ISSUE TITLE
Enforce a minimum contrast level

### ISSUE THREAD

**Issue by jaredly** [xtermjs/xterm.js#1514]
(I'm on VSCode 1.24.1)
iTerm2 has this really nice feature where you can enforce a minimum level of contrast, which is invaluable to making sure that all text shows up, regardless of color scheme / background color. 
![image](https://user-images.githubusercontent.com/112170/41439614-b2381766-6fe8-11e8-8988-f63c666d0fb1.png)

For example, the text to the right is basically unreadable, unless I select it, in which case it becomes marginally readable.
![image](https://user-images.githubusercontent.com/112170/41439604-a6e1bb6a-6fe8-11e8-906f-75184393c55a.png)

It would be awesome to have an option to enforce a minimum level of contrast.

It looks like https://github.com/xtermjs/xterm.js/blob/4717619f762dd62397ef21363934a19ecf1d8be1/src/renderer/BaseRenderLayer.ts#L240 would be the place to do it? although I guess it might need some changes to the CharAtlast as well

**Comment by Tyriar** [comment]
I guess this could be an opt-in setting which overrides the foreground color and if necessary the background color too?

Foreground and background are done in 2 passes so it might get tricky how to influence both. Perhaps only changing one or the other would be better? This should be done in the `TextRenderLayer`:

https://github.com/xtermjs/xterm.js/blob/4717619f762dd62397ef21363934a19ecf1d8be1/src/renderer/TextRenderLayer.ts#L214

https://github.com/xtermjs/xterm.js/blob/4717619f762dd62397ef21363934a19ecf1d8be1/src/renderer/TextRenderLayer.ts#L196

**Comment by jaredly** [comment]
I think that iterm2 only modifies the foreground color, and I think that should be sufficient
